src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ConflictResolution.targets

```diff
@@ -0,0 +1,87 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.ConflictResolution.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved. 
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+
+    <!-- The condition on this target causes it to be skipped during design-time builds if
+        the restore operation hasn't run yet.  This is to avoid displaying an error in
+        the Visual Studio error list when a project is created before NuGet restore has
+        run and created the assets file. -->
+  <Target Name="_GetLockFileAssemblies"
+          DependsOnTargets="ComputePrivateAssetsPackageReferences;
+                            _DefaultMicrosoftNETPlatformLibrary"
+          Condition=" '$(DesignTimeBuild)' != 'true' Or Exists('$(ProjectAssetsFile)')">
+    <!-- Essentially a copy of the SDKs RunResolvePublishAssemblies.
+         We need to find all the files that will be loaded from deps for conflict resolution.
+         Note that the RuntimeIdentifier is not passed to this task, as we want to process the
+         rid-less assets which will be passed to the compiler.
+         -->
+    <ResolvePublishAssemblies ProjectPath="$(MSBuildProjectFullPath)"
+                              AssetsFilePath="$(ProjectAssetsFile)"
+                              TargetFramework="$(TargetFrameworkMoniker)"
+                              RuntimeIdentifier=""
+                              PlatformLibraryName="$(MicrosoftNETPlatformLibrary)"
+                              PrivateAssetsPackageReferences="@(PrivateAssetsPackageReference)">
+
+      <Output TaskParameter="AssembliesToPublish" ItemName="_LockFileAssemblies" />
+
+    </ResolvePublishAssemblies>
+  </Target>
+
+  <UsingTask TaskName="HandlePackageFileConflicts" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+
+  <PropertyGroup>
+    <!-- If SelfContained is not true the host will run from packages,
+             but we still need to consider those package items for conflicts -->
+    <HandlePackageFileConflictsDependsOn Condition="'$(SelfContained)' != 'true'">_GetLockFileAssemblies</HandlePackageFileConflictsDependsOn>
+  </PropertyGroup>
+  
+  <Target Name="HandlePackageFileConflicts" AfterTargets="ResolvePackageDependenciesForBuild" DependsOnTargets="$(HandlePackageFileConflictsDependsOn)">
```

Comment:
I do think folks will depend on doing stuff after these, so we have to imagine an extension point for "where do I sequence a target to get a filtered list of reference/referencecopylocalpaths/deps-file-items.

I agree with @eerhardt that these should not be public.

They should also not occupying the "good spot" for sequencing.  What I mean here is that folks should easily be able to run after them without having ambiguous ordering with these targets themselves.

The names/sequencing they have now are due to me implementing them outside the SDK.  Inside the SDK I think they should be rolled into other target sequences.

Here's a straw man:
1. As a dev extending the SDK I can sequence a target AfterTargets="ResolveReferences" and I am garunteed to get a complete and filtered list of Reference and ReferencePath items.
2. As a dev extending the SDK I can sequence a target AfterTargets="ResolvePackageDependenciesForBuild" and I am garunteed to get a complete and filtered list of items used to construct the build-time deps file.
3. As a dev extending the SDK I can sequence a target AfterTargets="RunResolvePublishAssemblies" and I am garunteed to get a complete and filtered list of ResolvedAssembliesToPublish items (not the case today).
4. As a dev extending the SDK I can sequence a target AfterTargets="ComputeFilesToPublish" and I am garunteed to get a complete and filtered list of ResolvedFileToPublish items, as well as a complete and filtered set of items that will be used to construct the deps file.

Perhaps 1 & 2 and 3 & 4 can be collapsed so long as the items output don't have a relationship (eg one copied to the other).

Moreover, we should have a property that folks can append Targets to so that folks that want to hook this sequence can insert themselves into a deterministic sequence and follow a pattern with common inputs/outputs (a specific set of items) that they can operate on and garuntee those items are the only thing they need to care about updating.